### DISCUSSION TITLE
Valid liquid code causes TokenizationError

### DISCUSSION THREAD

**Discussion by b-wall** [harttle/liquidjs#630] [Working with Shopify]
The following code breaks the LiquidJS parser:

` <a class="text-black underline hover:text-gray-300" href="http://www.google.com/maps/search/?api=1&query={{ location.name }} - {{ [location.address1, location.address2].compact.join(', ') }}, {{ location.city }}, {{ location.province }}, {{ location.zip }}" target="_blank">
    Open map&nbsp;
    <span class="small">&rarr;</span>
</a>`

More specifically, this part: `{{ [location.address1, location.address2].compact.join(', ') }}`

Although this is not common liquid, its actually used in Shopify's official 'ready to pickup' email template. 

Is there a way you could cause this type of code not to throw an error? Thank you.

**Comment by harttle** [comment]
I don't recognize `{{ [location.address1, location.address2].compact.join(', ') }}` as liquid syntax. It has to be something like:

```liquid
{{ arr | compact | join: ',' }}
```

Could you add a link to the snippet so we can have a better understanding?

**Reply by b-wall** [reply]
Thanks for responding so quickly. If you've seen the Shopify email templates, they often use liquid syntax that's not "official" and capable of being used by everyone. Nonetheless, its still being parsed correctly when emails are compiled and sent out. 

I don't have a link to the snippet, but if you go to any Shopify store > settings > notifications > 'ready to pickup' email template

You can see it there. It's the default code. Line 121
